src/d3a/d3a_core/global_objects.py

```diff
@@ -15,29 +15,72 @@
+        uppermost level of the tree
+        """
+        # the lazy import is needed in order to avoid circular imports
+        from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+        for child in area.children:
+            if isinstance(child.strategy, InfiniteBusStrategy):
+                self.current_feed_in_tariff = \
+                    find_object_of_same_weekday_and_time(child.strategy.energy_buy_rate,
+                                                         current_market_slot)
+                return
+
+    def buffer_market_maker_rate(self):
+        if self.root_area.current_market:
+            self.current_market_maker_rate = get_market_maker_rate_from_config(
+                                                      self.root_area.current_market)
+
+    def update(self, market_cycle=False):
+        if self.root_area.current_market is None:
+            return
+        self._create_grid_tree_dict(self.root_area, self.area_stats_tree_dict)
+        if market_cycle:
+            self._buffer_feed_in_tariff(self.root_area, self.root_area.current_market.time_slot)
+            self.buffer_market_maker_rate()
+
+    def is_it_time_for_external_tick(self, current_tick_in_slot):
+        return self.external_tick_counter.is_it_time_for_external_tick(current_tick_in_slot)
 
     def _create_grid_tree_dict(self, area, outdict):
-        outdict[area.name] = {}
+        # the lazy import is needed in order to avoid circular imports
+        from d3a.models.strategy.external_strategies import ExternalMixin
+        outdict[area.uuid] = {}
         if area.children:
             if area.current_market:
-                area_dict = {"last_market_slot": area.current_market.time_slot_str,
-                             "last_market_bill": area.stats.get_last_market_stats_for_grid_tree(),
-                             "last_market_stats": area.stats.get_price_stats_current_market(),
-                             "last_market_fee": area.current_market.fee_class.grid_fee_rate,
-                             "next_market_fee": area.get_grid_fee(),
-                             "children": {}}
+                area_dict = {'last_market_bill': area.stats.get_last_market_stats_for_grid_tree(),
+                             'last_market_stats': area.stats.get_price_stats_current_market(),
+                             'last_market_fee': area.current_market.fee_class.grid_fee_rate,
+                             'current_market_fee': area.get_grid_fee(),
+                             'area_name': area.name,
+                             'children': {}}
             else:
-                area_dict = {"children": {}}
-            outdict[area.name].update(area_dict)
+                area_dict = {'children': {},
+                             'area_name': area.name}
+            outdict[area.uuid].update(area_dict)
             for child in area.children:
```

Comment:
A side note for the future, we should decide on whether we should use recursion or use the area traversal function for such cases

Reply:
Please keep in mind that I have to create an outdict in the same structure as the indict here. Not sure whether this is possible with a traversal function